js/battle.js

```diff
@@ -5679,12 +5680,14 @@ var Battle = (function () {
 			break;
 		case 'join':
 		case 'j':
+			this.users[toUserid(args[1])] = ' ' + args[1];
 			if (!this.ignoreSpects) {
 				this.log('<div class="chat"><small>' + Tools.escapeHTML(args[1]) + ' joined.</small></div>', preempt);
 			}
 			break;
 		case 'leave':
 		case 'l':
+			delete this.users[toUserid(args[1])];
```

Comment:
Yes, that's my point, though, there's no need to keep track of users if the battle doesn't have a room attached.